src/subscription_manager/i18n.py

```diff
@@ -51,10 +54,100 @@ def configure_gettext():
     gettext.bind_textdomain_codeset(APP, 'UTF-8')
     locale.bind_textdomain_codeset(APP, 'UTF-8')
 
+
 translation = gettext.translation(APP, fallback=True)
 if six.PY3:  # gettext returns unicode in Python 3
     ugettext = translation.gettext
     ungettext = translation.ngettext
 else:
     ugettext = translation.ugettext
     ungettext = translation.ungettext
+
+
+class Locale(object):
+    """
+    Class used for changing languages on the fly
+    """
+
+    languages = {}
+    converter = {}
+
+    @classmethod
+    def set_gettext_method(cls, language):
+        """
+        Set gettext method and locale for specified language. This method is
+        intended for changing language on the fly, because rhsm service can
+        be used by many users with different language preferences at the
+        same time.
+        :param language: String representing locale
+                (e.g. de, de_DE, de_DE.utf-8, de_DE.UTF-8)
+        """
+        global ugettext, ungettext
+        lang = None
+
+        if language != '':
```

Comment:
The nesting of `if` statements is quite messy here.

Reply:
I agree. I will do refactoring of the code.